gate-web/src/main/groovy/com/netflix/spinnaker/gate/security/x509/X509AuthenticationUserDetailsService.groovy

```diff
@@ -68,14 +71,19 @@ class X509AuthenticationUserDetailsService implements AuthenticationUserDetailsS
 
     def roles = []
     if (email) {
+      log.debug("Adding email {} to roles.", email)
       roles.add(email)
     }
     if (rolesExtractor) {
-      roles += rolesExtractor.fromCertificate(x509)
+      def extractedRoles = rolesExtractor.fromCertificate(x509)
+      log.debug("Extracting roles {}", extractedRoles)
+      roles += extractedRoles
+    } else {
+      log.debug("Role extractor not defined.")
+      permissionService.login(email as String)
```

Comment:
Why wouldn't this be outside of the conditional? If we're using x509 certificates, we'll still want to have it interact with fiat.

Reply:
Ah we were wondering about this, from talking to @ttomsu it didn't seem like x509 would have to interact with fiat, because the groups would be passed later down the line. we can move this out of the conditional statement.

Reply:
I'm confused by this - by moving this in the (negative) conditional, (in the positive conditional) your extracting roles, but not telling Fiat about them. It seems like in the positive branch of the conditional you'll need to call `loginWithRoles`. Do I understand that correctly?